renku_notebooks/util/kubernetes_.py

```diff
@@ -140,8 +151,8 @@ def get_server_url(pod):
 
     pods = get_user_server_pods(user)
 
-    servers = {
-        pod.metadata.annotations["hub.jupyter.org/servername"]: {
+    servers = [
+        {
```

Comment:
use the name as key - then in `get_user_server` you can just extract it immediately.